catalog/db_branch_scanner.go

```diff
@@ -35,9 +42,37 @@ func NewDBBranchScanner(tx db.Tx, branchID int64, commitID CommitID, opts *DBSca
 		s.opts.BufferSize = DBScannerDefaultBufferSize
 	}
 	s.buf = make([]*DBScannerEntry, 0, s.opts.BufferSize)
+	commitsWhere, _ := getRelevantCommitsCondition(tx, branchID, commitID)
+	s.commitsWhere = commitsWhere
 	return s
 }
 
+func getRelevantCommitsCondition(tx db.Tx, branchID int64, commitID CommitID) (string, error) {
+	var branchMaxCommitID CommitID
+	var commits []string
+	var commitsWhere string
+	if commitID == UncommittedID || commitID == CommittedID {
+		branchMaxCommitID = MaxCommitID
+	} else {
+		branchMaxCommitID = commitID
+	}
+	// commit_id name is changed so that sorting will be performed on the numeric value, not the string value (where "10" is less than "2")
+	sql := "SELECT commit_id::text as str_commit_id FROM catalog_commits WHERE branch_id = $1 AND commit_id <= $2 ORDER BY commit_id"
```

Comment:
1. why do we need the string value we can just use the value in numeric golang slice and give it to our builder.
2. missing the limit or limit +1 in order not to read more than we need